From patchwork Thu Feb 20 06:19:35 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Manorit Chawdhry <m-chawdhry@ti.com>
X-Patchwork-Id: 24821
X-Patchwork-Delegate: praneeth@ti.com
From: Manorit Chawdhry <m-chawdhry@ti.com>
Date: Thu, 20 Feb 2025 11:49:35 +0530
Subject: [PATCH tiU25.01] TI: HACK: Kconfig: Add default bootcmd for all
 devices of ARCH_K3
Message-ID: <20250220-b4-internal-legacy-bootmeth-v1-1-e074c9306466@ti.com>
Return-Path: nm@ti.com
List-Post: <mailto:linux-patch-review@list.ti.com>
MIME-Version: 1.0
To: linux-patch-review@list.ti.com
Cc: Udit Kumar <u-kumar1@ti.com>, Praneeth Bajjuri <praneeth@ti.com>, Vignesh
 Raghavendra <vigneshr@ti.com>, Manorit Chawdhry <m-chawdhry@ti.com>
List-ID: <linux-patch-review@list.ti.com>
List-Post: <mailto:linux-patch-review@list.ti.com>

Till we have the standard boot flow in place with boot.scr, temporarily
migrate to legacy bootmethod to allow developing without modifying grub
cfgs.

Signed-off-by: Manorit Chawdhry <m-chawdhry@ti.com>
---
J721e SD boot with the patch:
http://serenity.dal.design.ti.com:7777/wugakufayi.yaml
---
 boot/Kconfig | 1 +
 1 file changed, 1 insertion(+)


---
base-commit: 4d6c49e808b86832ba198e8884226e2ee959d229
change-id: 20250220-b4-internal-legacy-bootmeth-d55bfffe6793

Best regards,

diff --git a/boot/Kconfig b/boot/Kconfig
index 929a11c90e97b2eb4f0bfefe6be7740dce8eed5e..b6c94ce394097cdad9cf14a79a3d81e016a16d3c 100644
--- a/boot/Kconfig
+++ b/boot/Kconfig
@@ -1756,6 +1756,7 @@ config USE_BOOTCOMMAND
 config BOOTCOMMAND
 	string "bootcmd value"
 	depends on USE_BOOTCOMMAND && !USE_DEFAULT_ENV_FILE
+	default "run envboot; run bootcmd_ti_mmc; bootflow scan -lb" if ARCH_K3
 	default "bootflow scan -lb" if BOOTSTD_DEFAULTS && CMD_BOOTFLOW_FULL
 	default "bootflow scan" if BOOTSTD_DEFAULTS && !CMD_BOOTFLOW_FULL
 	default "run distro_bootcmd" if !BOOTSTD_BOOTCOMMAND && DISTRO_DEFAULTS
